pkg/cmd/server/api/v1/types.go

```diff
@@ -543,10 +543,12 @@ type RoutingConfig struct {
 type MasterNetworkConfig struct {
 	// NetworkPluginName is the name of the network plugin to use
 	NetworkPluginName string `json:"networkPluginName"`
-	// ClusterNetworkCIDR is the CIDR string to specify the global overlay network's L3 space
-	ClusterNetworkCIDR string `json:"clusterNetworkCIDR"`
-	// HostSubnetLength is the number of bits to allocate to each host's subnet e.g. 8 would mean a /24 network on the host
-	HostSubnetLength uint32 `json:"hostSubnetLength"`
+	// Deprecated and maintained for backwards compatibility, use ClusterNetworks instead.
+	DeprecatedClusterNetworkCIDR string `json:"clusterNetworkCIDR,omitempty"`
+	// ClusterNetworks is a list of cluster networks that defines the global overlay network's L3 space
+	ClusterNetworks []ClusterNetworkEntry `json:"clusterNetworks"`
```

Comment:
You need to update the documentation to describe the format. Godoc on fields IS our API doc.

The field names are fine, but you need to update the godoc to be correct and complete before you get approved.

Reply:
@smarterclayton how is:

DeprecatedClusterNetworkCIDR is the CIDR string to specify the global overlay network's L3 space.  Deprecated, but maintained for backwards compatibility, use ClusterNetworks instead.

DeprecatedHostSubnetLength is the number of bits to allocate to each host's subnet e.g. 8 would mean a /24 network on the host.  Deprecated, but maintained for backwards compatibility, use ClusterNetworks instead.

ClusterNetworks is a list of ClusterNetwork objects that defines the global overlay network's L3 space by specifying a set of CIDR and netmasks that the SDN can allocate addressed from.  If this is specified, then DeprecatedClusterNetworkCIDR and DeprecatedHostSubnetLength may not be set.